pkg/frontend/adminactions/resources_list.go

```diff
@@ -16,24 +17,39 @@ import (
 	"github.com/Azure/ARO-RP/pkg/util/subnet"
 )
 
-func (a *azureActions) ResourcesList(ctx context.Context) ([]byte, error) {
+var comaAsByteArray = []byte(",")
+
+func (a *azureActions) GroupResourceList(ctx context.Context) ([]mgmtfeatures.GenericResourceExpanded, error) {
 	clusterRGName := stringutils.LastTokenByte(a.oc.Properties.ClusterProfile.ResourceGroupID, '/')
 
-	resources, err := a.resources.ListByResourceGroup(ctx, clusterRGName, "resourceType ne 'Microsoft.Compute/snapshots'", "", nil)
-	if err != nil {
-		return nil, err
-	}
+	return a.resources.ListByResourceGroup(ctx, clusterRGName, "resourceType ne 'Microsoft.Compute/snapshots'", "", nil)
+}
 
+func (a *azureActions) ResourcesList(ctx context.Context, resources []mgmtfeatures.GenericResourceExpanded, writer io.WriteCloser) error {
+	defer writer.Close()
+
+	clusterRGName := stringutils.LastTokenByte(a.oc.Properties.ClusterProfile.ResourceGroupID, '/')
+	writer.Write([]byte("["))
 	// +4 because we expect +2 subnet and +1 vnets and optional +1 diskEncryptionSet
 	armResources := make([]arm.Resource, 0, len(resources)+4)
-	armResources, err = a.appendAzureNetworkResources(ctx, armResources)
+	armResources, err := a.appendAzureNetworkResources(ctx, armResources)
 	if err != nil {
 		a.log.Warnf("error when getting network resources: %s", err)
 	}
 	armResources, err = a.appendAzureDiskEncryptionSetResources(ctx, armResources)
 	if err != nil {
 		a.log.Warnf("error when getting DiskEncryptionSet resources: %s", err)
 	}
+	hasWritten := false
+
+	for _, resource := range armResources {
+		if hasWritten {
+			writer.Write(comaAsByteArray)
+		}
+		bytes, _ := resource.MarshalJSON()
+		writer.Write(bytes)
```

Comment:
Could we leverage your `writeObject` function here as well?

Reply:
good catch, will fix that tomorrow